main.go

```diff
@@ -579,3 +582,29 @@ func setLoggerForProduction(encoder zapcore.LevelEncoder, dest io.Writer) {
 	ctrl.SetLogger(newlogger)
 	klog.SetLogger(newlogger)
 }
+
+// setupSignalHandler registers a signal handler for SIGTERM and SIGINT and
+// returns a context which is canceled when one of these signals is caught after
+// waiting for the specified duration. If a second signal is caught then we
+// terminate immediately with exit code 1. The implementation has been stolen
```

Comment:
To avoid misinterpretation, let's avoid "stolen". "This augments controller-runtime's model to support the delay".

Also, have you considered submitting this to controller-runtime? Generally it's better to avoid forks when possible.

Reply:
Thank you for contributing!